go/inbloom/bloom.go

```diff
@@ -161,17 +161,29 @@ func (f *BloomFilter) Marshal() []byte {
 	return buf.Bytes()
 }
 
-// MarshalBase64 is a convenience method that dumps the filter's data to a base64 encoded string,
-// ready to be passed as an GET/POST parameter
-func (f *BloomFilter) MarshalBase64() string {
-	return base64.URLEncoding.EncodeToString(f.Marshal())
+// MarshalBase64 is a convenience method that dumps the filter's data to a base64 encoded string.
+// By default uses URLEncoding which ready to be passed as a GET/POST parameter.
+// Pass an encoding param to use different encoding.
+func (f *BloomFilter) MarshalBase64(encoding ...*base64.Encoding) string {
```

Comment:
I don't like this pattern so much. 
How about adding func (f *BloomFilter) MarshalBase64Encoding(encoding *base64.Encoding) and have the old function just call this one with nil?